test/e2e/autoscaling_utils.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"strconv"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	consumptionTimeInSeconds = 30
+	sleepTime                = 30 * time.Second
+	requestSizeInMilicores   = 100
+)
+
+// ConsumingRC is a tool for testing. It helps create specified usage of CPU or memory (Warnig: memory not supported)
+// typical use case:
+// rc.ConsumeCPU(600)
+// // ... check your assumption here
+// rc.ConsumeCPU(300)
+// // ... check your assumption here
+type ConsumingRC struct {
+	name      string
+	framework *Framework
+	channel   chan int
+	stop      chan int
+}
+
+// creates new ConsumingRC
+func NewConsumingRC(name string, replicas int, framework *Framework) *ConsumingRC {
+	startService(framework.Client, framework.Namespace.Name, name, replicas)
+	rc := &ConsumingRC{
+		name:      name,
+		framework: framework,
+		channel:   make(chan int),
+		stop:      make(chan int),
+	}
+	go rc.makeConsumeCPURequests()
+	rc.ConsumeCPU(0)
+	return rc
+}
+
+// consumes given number of CPU
+func (rc *ConsumingRC) ConsumeCPU(milicores int) {
+	rc.channel <- milicores
+}
+
+func (rc *ConsumingRC) makeConsumeCPURequests() {
+	defer GinkgoRecover()
+	var count int
+	var rest int
+	for {
+		select {
+		case milicores := <-rc.channel:
+			count = milicores / requestSizeInMilicores
+			rest = milicores - count*requestSizeInMilicores
+		case <-time.After(sleepTime):
+			rc.sendConsumeCPUrequests(count, requestSizeInMilicores, consumptionTimeInSeconds)
```

Comment:
Nit: please invoke it only if count or rest >0. Right now it is confusing what will happen if there is no millicores is set.